typings/tests.ts

```diff
@@ -484,6 +489,31 @@ client.on('messageCreate', ({ channel }) => {
   channel.send();
   // @ts-expect-error
   channel.send({ another: 'property' });
+
+  // Check collector creations.
+
+  // Verify that buttons interactions are inferred.
+  const buttonCollector = message.createMessageComponentCollector({ componentType: 'BUTTON' });
+  assertType<Promise<InteractionCollector<ButtonInteraction>>>(
+    message.awaitMessageComponent({ componentType: 'BUTTON' }),
+  );
+  assertType<InteractionCollector<ButtonInteraction>>(buttonCollector);
+
+  // Verify that select menus interaction are inferred.
+  const selectMenuCollector = message.createMessageComponentCollector({ componentType: 'SELECT_MENU' });
+  assertType<Promise<InteractionCollector<SelectMenuInteraction>>>(
+    message.awaitMessageComponent({ componentType: 'SELECT_MENU' }),
+  );
+  assertType<InteractionCollector<SelectMenuInteraction>>(selectMenuCollector);
+
+  // Verify that message component interactions are default collected types.
+  const defaultCollector = message.createMessageComponentCollector();
+  assertType<Promise<InteractionCollector<MessageComponentInteraction>>>(message.awaitMessageComponent());
+  assertType<InteractionCollector<MessageComponentInteraction>>(defaultCollector);
+
+  // Verify that additional options don't affect default collector types.
+  const semiDefaultCollector = message.createMessageComponentCollector({ interactionType: 'APPLICATION_COMMAND' });
```

Comment:
Doesn't this throw an error / shouldn't it throw a type error?

Reply:
So `interactionType` is an allowed option currently for `InteractionCollectorOptions` which it seems that `createMesaageComponent()` uses, IIRC thats how it was setup before this PR.

But I do see your point, this may be a mistake, since interaction type doesn't have anything to do with component interactions.